### PR TITLE
graphql_directives:  type resolution system

### PR DESCRIPTION
**Pull request by pmelab**
## Package(s) involved

`amazeelabs/graphql_directives`

## Description of changes

Add directive-based type resolution system (see README.md).

## Motivation and context

Reuse pluggable directives for interface and union type resolution.
Further increase reusability.

## Related Issue(s)

#1169

## How has this been tested?

* unit tests
* kernel tests

### CODE REVIEW

**Comment by pmelab** [comment]
Current dependencies on/for this PR:
* development
  * **PR #1228** <a href="https://app.graphite.dev/github/pr/AmazeeLabs/silverback-mono/1228" target="_blank">Graphite</a> 
    * **PR #1229** <a href="https://app.graphite.dev/github/pr/AmazeeLabs/silverback-mono/1229" target="_blank">Graphite</a> 
      * **PR #1230** <a href="https://app.graphite.dev/github/pr/AmazeeLabs/silverback-mono/1230" target="_blank">Graphite</a> 
        * **PR #1231** <a href="https://app.graphite.dev/github/pr/AmazeeLabs/silverback-mono/1231" target="_blank">Graphite</a> 
          * **PR #1232** <a href="https://app.graphite.dev/github/pr/AmazeeLabs/silverback-mono/1232" target="_blank">Graphite</a> 
            * **PR #1233** <a href="https://app.graphite.dev/github/pr/AmazeeLabs/silverback-mono/1233" target="_blank">Graphite</a>  👈
              * **PR #1234** <a href="https://app.graphite.dev/github/pr/AmazeeLabs/silverback-mono/1234" target="_blank">Graphite</a> 
                * **PR #1244** <a href="https://app.graphite.dev/github/pr/AmazeeLabs/silverback-mono/1244" target="_blank">Graphite</a> 
                  * **PR #1249** <a href="https://app.graphite.dev/github/pr/AmazeeLabs/silverback-mono/1249" target="_blank">Graphite</a> 
                    * **PR #1245** <a href="https://app.graphite.dev/github/pr/AmazeeLabs/silverback-mono/1245" target="_blank">Graphite</a> 
                      * **PR #1250** <a href="https://app.graphite.dev/github/pr/AmazeeLabs/silverback-mono/1250" target="_blank">Graphite</a> 
                        * **PR #1255** <a href="https://app.graphite.dev/github/pr/AmazeeLabs/silverback-mono/1255" target="_blank">Graphite</a> 
                          * **PR #1257** <a href="https://app.graphite.dev/github/pr/AmazeeLabs/silverback-mono/1257" target="_blank">Graphite</a> 
                            * **PR #1267** <a href="https://app.graphite.dev/github/pr/AmazeeLabs/silverback-mono/1267" target="_blank">Graphite</a> 

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/AmazeeLabs/silverback-mono/1233?utm_source=stack-comment).

**Comment by pmelab** [comment]
Current dependencies on/for this PR:
* development
  * **PR #1228** <a href="https://app.graphite.dev/github/pr/AmazeeLabs/silverback-mono/1228" target="_blank">Graphite</a> 
    * **PR #1229** <a href="https://app.graphite.dev/github/pr/AmazeeLabs/silverback-mono/1229" target="_blank">Graphite</a> 
      * **PR #1230** <a href="https://app.graphite.dev/github/pr/AmazeeLabs/silverback-mono/1230" target="_blank">Graphite</a> 
        * **PR #1231** <a href="https://app.graphite.dev/github/pr/AmazeeLabs/silverback-mono/1231" target="_blank">Graphite</a> 
          * **PR #1232** <a href="https://app.graphite.dev/github/pr/AmazeeLabs/silverback-mono/1232" target="_blank">Graphite</a> 
            * **PR #1233** <a href="https://app.graphite.dev/github/pr/AmazeeLabs/silverback-mono/1233" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/AmazeeLabs/silverback-mono/1233?utm_source=stack-comment).

File: packages/composer/amazeelabs/graphql_directives/README.md [hunk 1]
```diff
@@ -103,6 +127,23 @@ type Query {
...
+union type resolution.
+
+```graphql
+union Letters @prop(key: "type") = A | B
+
+type A @type(id: "a") {
+  type: String!
+}
+
+type B @type(id: "B") {
```
Comment (Leksat) [inline packages/composer/amazeelabs/graphql_directives/README.md]: Should it be `id: "b"` ? (lowercase)

**Review by Leksat — COMMENTED** [review]
How does it replace

```
@resolveProperty(path: "field_text.0.processed")
@resolveProperty(path: "field_entity_reference.entity")
```

?

File: packages/composer/amazeelabs/graphql_directives/README.md (hunk 1 above)
Comment (pmelab) [reply packages/composer/amazeelabs/graphql_directives/README.md]: > How does it replace
> 
> ```
> @resolveProperty(path: "field_text.0.processed")
> @resolveProperty(path: "field_entity_reference.entity")
> ```
> 
> ?

Not at all. My plan is to move `@resolveProperty` (and most other directives we have) to plugins to make them more maintainable.

**Review by chindris — APPROVED** [review]

**Comment by pmelab** [comment]
❌ This merge job failed. This pull request was not merged
Stack job ID: tLYBbcASzYVOW56lDOXe.
See details on [graphite.dev](https://app.graphite.dev/github/pr/AmazeeLabs/silverback-mono/1233)

**Pull request merged**